single_include/nlohmann/json.hpp

```diff
@@ -5542,30 +5543,46 @@ class iterator_input_adapter
     using char_type = typename std::iterator_traits<IteratorType>::value_type;
 
     iterator_input_adapter(IteratorType first, IteratorType last)
-        : current(std::move(first)), end(std::move(last))
+        : current(std::move(first)), end(std::move(last)), current_has_been_consumed(false)
     {}
 
     typename std::char_traits<char_type>::int_type get_character()
     {
-        if (JSON_HEDLEY_LIKELY(current != end))
+
+        if (JSON_HEDLEY_LIKELY(current_has_been_consumed))
         {
-            auto result = std::char_traits<char_type>::to_int_type(*current);
             std::advance(current, 1);
-            return result;
         }
 
-        return std::char_traits<char_type>::eof();
+        if (JSON_HEDLEY_LIKELY(current != end))
+        {
+            current_has_been_consumed = true;
+            return std::char_traits<char_type>::to_int_type(*current);
+        }
+        else
+        {
+            current_has_been_consumed = false;
+            return std::char_traits<char_type>::eof();
+        }
+
     }
 
   private:
-    IteratorType current;
-    IteratorType end;
+    mutable IteratorType current;
+    const IteratorType end;
+    mutable bool current_has_been_consumed;
```

Comment:
```
/__w/json/json/include/nlohmann/detail/input/input_adapters.hpp:163:18: error: use default member initializer for 'current_has_been_consumed' [modernize-use-default-member-init,-warnings-as-errors]
    mutable bool current_has_been_consumed;
                 ^
                                          {false}
```

clang-tidy seems to insist on this:
`mutable bool current_has_been_consumed{false};`

(Didn't know/remember that.)

Reply:
Please use

```cpp
mutable bool current_has_been_consumed = false;
```

This library has a troubled relationship to braced initializers... :-)